bundles/action/org.openhab.action.twitter/src/main/java/org/openhab/action/twitter/internal/Twitter.java

```diff
@@ -57,21 +64,104 @@ public static boolean sendTweet(@ParamDoc(name = "tweetTxt", text = "the Tweet t
+            if (fileToAttach != null && fileToAttach.isFile()) {
+            	status.setMedia(fileToAttach);
+            }
+            final Status updatedStatus = client.updateStatus(status);
+            logger.debug("Successfully sent Tweet '{}'", updatedStatus.getText());
             return true;
         } catch (TwitterException e) {
-            logger.error("Failed to send Tweet '" + tweetTxt + "' because of: " + e.getLocalizedMessage());
+            logger.error("Failed to send Tweet '" + abbreviatedTweetTxt + "' because of: " + e.getLocalizedMessage());
             return false;
         }
     }
 
     /**
+     * Sends a Tweet via Twitter
+     * 
+     * @param tweetTxt the Tweet to send
+     * 
+     * @return <code>true</code>, if sending the tweet has been successful and
+     *         <code>false</code> in all other cases.
+     */
+    @ActionDoc(text = "Sends a Tweet via Twitter", returns = "<code>true</code>, if sending the tweet has been successful and <code>false</code> in all other cases.")
+    public static boolean sendTweet(@ParamDoc(name = "tweetTxt", text = "the Tweet to send") String tweetTxt) {
+    	if (! checkPrerequesites()) {
+    		return false;
+    	}
+        return sendTweet(tweetTxt, null);
+    }
+
+    /**
+     * Sends a Tweet via Twitter
+     * 
+     * @param tweetTxt the Tweet to send
+     * 
+     * @return <code>true</code>, if sending the tweet has been successful and
+     *         <code>false</code> in all other cases.
+     */
+    @ActionDoc(text = "Sends a Tweet via Twitter", returns = "<code>true</code>, if sending the tweet has been successful and <code>false</code> in all other cases.")
+    public static boolean sendPicture(@ParamDoc(name = "tweetTxt", text = "the Tweet to send") String tweetTxt, @ParamDoc(name = "tweetPicture", text = "the picture to attach") String tweetPicture) {
+    	if (! checkPrerequesites()) {
+    		return false;
+    	}
+
+        // prepare the image attachment
+        File fileToAttach = null;
+        boolean deleteTemporaryFile = false;
+        if (StringUtils.startsWith(tweetPicture, "http")) {
+    		final String tDir = System.getProperty("java.io.tmpdir"); 
+    		final String path = tDir + File.separator + "openhab-twitter-remote_attached_file" + "." + FilenameUtils.getExtension(tweetPicture); 
+        	try {
+        		final URL url = new URL(tweetPicture); 
+        		fileToAttach = new File(path); 
+        		// fileToAttach.deleteOnExit();  // good idea ? could lead to temporary files staying around until JVM is down
+        		deleteTemporaryFile = true;
+        		FileUtils.copyURLToFile(url, fileToAttach);
+        	} catch (final MalformedURLException e) {
+        		logger.error("Can't read file from '" + tweetPicture + "'", e);
+        	} catch (final IOException e) {
+        		logger.error("Can't save file from '" + tweetPicture + "' to '" + path + "'", e);
```

Comment:
This should be a WARN instead of an ERROR.
Also, need to use parameterized logging instead of string concatenation.